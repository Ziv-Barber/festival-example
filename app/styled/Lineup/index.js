// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars
import styled from 'styled-components'

// Other components that we need:
import { ItemsList } from '../../components/ItemsList'
import ActBox from '../ActBox'

// Content that we need:
import lineupContent from '../../content/lineup'

// Create the list of items for the Lineup component:
const lineupItems = lineupContent.map((value) => ({
  key: value.key,
  component: ActBox,
  compProps: {
    name: value.name,
    image: value.image
  }
  // gridProps
}))

/**
 * Lineup react component.
 */
export default styled(ItemsList).attrs({
  items: lineupItems,
  caption: 'Lineup',
  captionProps: '',
  captionVariant: '',
  containerId: '',
  gridRootDefProps: {
    spacing: 8,
    justify: 'space-around',
    alignItems: 'center'
  },
  gridDefProps: {
    xs: 12,
    sm: 4,
    md: 3,
    lg: 2,
    xl: 1
  }
})`
  margin: 0 !important;
  padding: ${(props) => props.theme.spacing.unit}px 0 !important;
  width: 100% !important;
  background: linear-gradient(to bottom, #cce0ff, #99c2ff);
  color: white;

  & .listHeader {
    padding: ${(props) => props.theme.spacing.unit}px !important;
    width: 100% !important;
    text-align: center;
    text-shadow: 2px 2px 5px #fff !important;
    color: #003366;
  }

  & .listGrid {
    margin: 0 !important;
    width: 100% !important;
    padding: ${(props) => props.theme.spacing.unit}px !important;
  }

  & .listItem {
    margin: ${(props) => props.theme.spacing.unit}px !important;
    z-index: 1050;
    display: flex;
    justify-content: center;
    align-items: center;
  }
`
