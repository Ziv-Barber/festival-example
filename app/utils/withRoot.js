// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars
import { MuiThemeProvider } from '@material-ui/core/styles'
import { ThemeProvider } from 'styled-components'
import CssBaseline from '@material-ui/core/CssBaseline'
import theme from '../theme'

/**
 * withRoot HOC to wrap a react component with the app's theme provider.
 */
const withRoot = (Component: React.ComponentType<any>) => {
  const withRootBase = (props: any) => (
    <MuiThemeProvider theme={theme}>
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <CssBaseline />
          <Component {...props} />
        </React.Fragment>
      </ThemeProvider>
    </MuiThemeProvider>
  )

  // $FlowFixMe
  withRootBase.displayName = `withRoot(${Component.displayName ||
    Component.name})`

  return withRootBase
}

export default withRoot
