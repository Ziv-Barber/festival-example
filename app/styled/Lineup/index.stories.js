import React from 'react' // eslint-disable-line no-unused-vars
import { storiesOf } from '@storybook/react'
import Lineup from './index'

storiesOf('Lineup', module).add(
  'basic',
  () => <Lineup text="Basic example" />,
  { notes: { markdown: 'This is an example to the Lineup component.' } }
)
