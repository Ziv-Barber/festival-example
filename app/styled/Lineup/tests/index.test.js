// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import { render, cleanup } from 'react-testing-library'

import Lineup from '../index'
import withRoot from '../../../utils/withRoot'

// For flow:
let { describe, it, afterEach, expect } = global

/** @test {Lineup} */
describe('<Lineup />', () => {
  afterEach(cleanup)

  /** @test {Lineup} */
  it('should work', () => {
    const App = withRoot(Lineup)
    const { comtainer } = render(<App />)
    expect(comtainer).toMatchSnapshot()
  })
})
