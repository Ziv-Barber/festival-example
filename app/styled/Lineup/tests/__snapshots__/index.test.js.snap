// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lineup /> should work 1`] = `undefined`;
